pkg/operator/controller/gatewayapi/crds.go

```diff
@@ -0,0 +1,128 @@
+package gatewayapi
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"github.com/openshift/cluster-ingress-operator/pkg/manifests"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// managedCRDs is a list of CRDs that this controller manages.
+var managedCRDs = []*apiextensionsv1.CustomResourceDefinition{
+	manifests.GatewayClassCRD(),
+	manifests.GatewayCRD(),
+	manifests.HTTPRouteCRD(),
+}
+
+// ensureCRD attempts to ensure that the specified CRD exists and returns a
+// Boolean indicating whether it exists, the CRD if it does exist, and an error
+// value.
+func (r *reconciler) ensureCRD(ctx context.Context, desired *apiextensionsv1.CustomResourceDefinition) (bool, *apiextensionsv1.CustomResourceDefinition, error) {
+	name := types.NamespacedName{
+		Namespace: desired.Namespace,
+		Name:      desired.Name,
+	}
+	have, current, err := r.currentCRD(ctx, name)
+	if err != nil && !errors.IsNotFound(err) {
+		return have, current, err
+	}
+
+	switch {
+	case !have:
+		if err := r.createCRD(ctx, desired); err != nil {
+			return false, nil, err
+		}
+		return r.currentCRD(ctx, name)
+	case have:
+		if updated, err := r.updateCRD(ctx, current, desired); err != nil {
+			return have, current, err
+		} else if updated {
+			return r.currentCRD(ctx, name)
+		}
+	}
+	return have, current, nil
```

Comment:
Is this dead code?

Reply:
We reach this line if `have` is true and no update was needed.